src/open_inwoner/accounts/signals.py

```diff
@@ -84,7 +84,10 @@ def _update_user_from_esuite(
 def _update_eherkenning_user_from_kvk_api(user: User):
     kvk_client = KvKClient()
 
-    vestiging = kvk_client.get_company_headquarters(kvk=user.kvk)
+    if user.vestiging:
+        kvk_client.get_vestiging(vestiging=user.vestiging)
+    else:
+        vestiging = kvk_client.get_company_headquarters(kvk=user.kvk)
```

Comment:
```suggestion
        vestiging = kvk_client.get_vestiging(vestiging=user.vestiging) if user.vestiging else kvk_client.get_company_headquarters(kvk=user.kvk)
```

(Both paths need to set `vestiging`, otherwise it's potentially undefined on `l.92`).